util_function_groups/column_operations.R

```diff
@@ -0,0 +1,137 @@
+        }
+        print("Validated datatype")
+    }
+    
+    validate_accepted_values <- function(column_attributes, x){
+        if(!all(x %in% column_attributes$accepted_values)){
+            util.warn(column_attributes$column %+% " has non-accepted values.")
+        }
+        print("Validated accepted values")
+    }
+    
+    validate_accepted_range <- function(column_attributes, x){
+        if(!is.numeric(x)){
+            util.warn(column_attributes$column %+% " is not numeric, but " %+%
+                          "you set min and max values for it.")
+        }
+        x_min <- min(column_attributes$accepted_range)
+        x_max <- max(column_attributes$accepted_range)
+        if(!all(x >= x_min & x <= x_max, na.rm = TRUE)){
+            util.warn(column_attributes$column %+%
+                          " has out of range values.")
+        }
+        print("Validated accepted range")
+    }
+    
+    validate_blanks_allowed <- function(column_attributes, x){
+        # if blanks are not allowed and there are any blanks, throw warning
+        if(!column_attributes$blanks_allowed & any(util.is_blank(x))){
+            util.warn(column_attributes$column %+% " has blank values, when none are allowed.")
+        }
+        print("Validated blanks allowed")
+    }
+    
+    validate_required <- function(column_attributes, df_names){
+        # make sure the "column_required" attribute exists
+        if(exists("column_attributes$column_required")){
+            # if it does, throw a warning if the column is required but it doesn't exist
+            if(column_attributes$column_required & !column_attributes$column %in% df_names){
+                util.warn("Required column " %+% column %+% " does not appear in the data.")
+            }
+        }
+        print("Validated required")
+    }
+    
+    for(column in names(column_validation)){
+        
+        # pick out the attributes of the column in question
+        column_attributes <- column_validation[[column]]
+        
+        # add the column name itself as an attribute of the column
+        column_attributes$column <- column
+        
+        # check whether the column exists
+        if(column %in% names(df)){
+            values <- df[,column]
+            # If it does, run the appropriate checks
+            for(attribute in names(column_attributes)){
+                switch(attribute,
+                       blanks_allowed = validate_blanks_allowed(column_attributes,values),
+                       accepted_values = validate_accepted_values(column_attributes,values),
+                       accepted_range = validate_accepted_range(column_attributes,values),
+                       datatype = validate_datatype(column_attributes,values)
+                )
+            }
+        }
+        validate_required(column_attributes, names(df))
+        # Finally, check that the column exists, if it's required
+        # If it was, throw a warning.
```

Comment:
Should this comment be above the function call on line 83? Right now it looks like the comment is talking about code which is missing...